org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/AbstractADWindowContent.java

```diff
@@ -2721,21 +2721,18 @@ public void onSaveCreate()
 			@Override
 			public void onCallback(Boolean result)
 			{
-				if(result)
-		    	{
-		    		adTabbox.getSelectedGridTab().dataRefreshAll(true, true);
-		    		adTabbox.getSelectedGridTab().refreshParentTabs();
-		    		IADTabpanel dirtyTabpanel = (IADTabpanel) Executions.getCurrent().removeAttribute("adtabpane.saved");
-		    		if (dirtyTabpanel != null && dirtyTabpanel.getGridTab().isDetail()) {
-		    			try {
-							adTabbox.getSelectedTabpanel().getDetailPane().onNew();
-						} catch (Exception e) {
-							throw new RuntimeException(e);
-						}
-		    		} else {
-		    			onNew();
-		    		}
-		    	}
+				adTabbox.getSelectedGridTab().dataRefreshAll(true, true);
+				adTabbox.getSelectedGridTab().refreshParentTabs();
+				IADTabpanel dirtyTabpanel = (IADTabpanel) Executions.getCurrent().removeAttribute("adtabpane.saved");
+				if (dirtyTabpanel != null && dirtyTabpanel.getGridTab().isDetail()) {
+					try {
+						adTabbox.getSelectedTabpanel().getDetailPane().onNew();
+					} catch (Exception e) {
+						throw new RuntimeException(e);
+					}
+				} else {
+					onNew();
+				}
```

Comment:
why remove the `if(result)` checking ?

Reply:
result is the return of the onSave operation.
The dataSave is returning ERROR when the record has not been modified, and there is nothing to save, and consequently the onSave is returning false, so the onNew is not called in this case.  I tested it and it seems to work without problems.

Reply:
Not working well for the following:
- Open Product Window
- Create new record, Enter Weeder for both Search Key and Name
- Press Alt+A to save

Reply:
Thanks @hengsin - I added a new commit that works better.